Fix scripts issues found by mypy

```
scripts/pass_rate.py:140: error: "ArgumentParser" has no attribute "reports"  [attr-defined]
scripts/pass_rate.py:141: error: "ArgumentParser" has no attribute "skiplist_dir"  [attr-defined]
scripts/pass_rate.py:146: error: "ArgumentParser" has no attribute "reports"  [attr-defined]
scripts/run_tutorial.py:44: error: Argument 1 to "module_from_spec" has incompatible type "ModuleSpec | None"; expected "ModuleSpec"  [arg-type]
scripts/run_tutorial.py:48: error: Item "None" of "ModuleSpec | None" has no attribute "loader"  [union-attr]
scripts/run_tutorial.py:48: error: Item "None" of "Loader | Any | None" has no attribute "exec_module"  [union-attr]
scripts/compare-ci-runs/compare_runs.py:132: error: Argument 2 to "merge_triton_xetla_reports_data" has incompatible type "str"; expected "Path"  [arg-type]
scripts/compare-ci-runs/compare_runs.py:132: error: Argument 3 to "merge_triton_xetla_reports_data" has incompatible type "str"; expected "Path"  [arg-type]
Found 8 errors in 3 files (checked 7 source files)
```

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -137,6 +137,13 @@ repos:
           - --disable=fixme
         stages: [pre-commit, pre-push, manual]
 
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.13.0
+    hooks:
+      - id: mypy
+        name: mypy for scripts
+        stages: [pre-commit, pre-push, manual]
+        files: '^scripts/.*'
 
 exclude: |
   (?x)(

--- a/scripts/compare-ci-runs/compare_runs.py
+++ b/scripts/compare-ci-runs/compare_runs.py
@@ -1,7 +1,6 @@
 #! /usr/bin/env python3
 
 import argparse
-import pathlib
 import shutil
 import subprocess
 import os

--- a/scripts/compare-ci-runs/compare_runs.py
+++ b/scripts/compare-ci-runs/compare_runs.py
@@ -1,6 +1,7 @@
 #! /usr/bin/env python3
 
 import argparse
+import pathlib
 import shutil
 import subprocess
 import os
@@ -111,9 +112,9 @@ def merge_triton_xetla_reports_data(config: str, triton_file: Path, xetla_file: 
         return pd.DataFrame()
 
 
-def build_triton_benchmark_reports_path(directory: Path, report_name: str) -> str:
+def build_triton_benchmark_reports_path(directory: Path, report_name: str) -> Path:
     """Construct the full file path for a given report name."""
-    return os.path.join(directory, "benchmark-reports", f"{report_name}-report.csv")
+    return directory / "benchmark-reports" / f"{report_name}-report.csv"
 
 
 def parse_triton_benchmark_data(config: str, directory: Path) -> pd.DataFrame:

--- a/scripts/pass_rate.py
+++ b/scripts/pass_rate.py
@@ -135,13 +135,13 @@ def find_stats(stats: List[ReportStats], name: str) -> ReportStats:
     raise ValueError(f'{name} not found')
 
 
-def parse_junit_reports(args: argparse.ArgumentParser) -> List[ReportStats]:
+def parse_junit_reports(args: argparse.Namespace) -> List[ReportStats]:
     """Parses junit report in the specified directory."""
     reports_path = pathlib.Path(args.reports)
     return [parse_report(report, args.skiplist_dir) for report in reports_path.glob('*.xml')]
 
 
-def parse_tutorials_reports(args: argparse.ArgumentParser) -> List[ReportStats]:
+def parse_tutorials_reports(args: argparse.Namespace) -> List[ReportStats]:
     """Parses tutorials reports in the specified directory."""
     reports_path = pathlib.Path(args.reports)
     stats = ReportStats(name='tutorials')
@@ -157,7 +157,7 @@ def parse_tutorials_reports(args: argparse.ArgumentParser) -> List[ReportStats]:
     return [stats]
 
 
-def parse_reports(args: argparse.ArgumentParser) -> List[ReportStats]:
+def parse_reports(args: argparse.Namespace) -> List[ReportStats]:
     """Parses all report in the specified directory."""
     return parse_junit_reports(args) + parse_tutorials_reports(args)
 

--- a/scripts/run_tutorial.py
+++ b/scripts/run_tutorial.py
@@ -41,6 +41,8 @@ def create_argument_parser() -> argparse.ArgumentParser:
 def run_tutorial(path: pathlib.Path):
     """Runs """
     spec = importlib.util.spec_from_file_location('__main__', path)
+    if not spec or not spec.loader:
+        raise AssertionError(f'Failed to load module from {path}')
     module = importlib.util.module_from_spec(spec)
     # set __file__ to the absolute name, a workaround for 10i-experimental-block-pointer, which
     # uses dirname of its location to find 10-experimental-block-pointer.

